### PR TITLE
'apictl logout' revoke the currently used token when an user is logging out. [Fix for 327]

### PR DESCRIPTION
**Pull request by Chamindu36**
## Purpose
When a user tries to logout from an environment the generated access token for that user is not revoking and it keeps there. This has to be fixed as when the user is logging out that token has to be revoked. Therefore this has to be fixed as when an user logging out used token should be revoked.

**Reproducing the issue**

1. User login using apictl login
2. Do some work
3. apictl logout
4. Same user logins using "apictl login"
5. When doing next operations, we can see that the same previous token is used for those.

## Goals
Fixes https://github.com/wso2/product-apim-tooling/issues/327

## Approach
**1. Create a function RevokeAccessToken()**
_This Function will create an HTTP post request to invoke token revocation endpoint and kill the current access token for that particular user before the user logs out from the environment._
**2. Use that method in logout.go**
_Before the user logout from the environment and erase credentials from .json file a newly introduced method will be executed to revoke the current access token._
**3. Provide tokenRevokeEndpoint.** 
_When providing this revoke token endpoint user does not need to do anything. Derivation of this revocation token endpoint will be done internally. That derived token revocation endpoint will be provided as the url RevokeAccessToken() function._

## User stories
Logging out from an environment in apictl.

## Documentation
No Document changes

## Test environment
Ubuntu 20.04
Java 1.8_252
APIM 3.2.0

### CODE REVIEW

File: import-export-cli/utils/envManagementUtils.go [hunk 1]
```diff
@@ -353,4 +353,15 @@ func GetInternalTokenEndpointOfEnv(env, filePath string ) string {
...
+//Get token endpoint for Token revocation
+//@return endpoint URL of token revocation endpoint
+func GetTokenRevokeEndpoint (env, filePath string) string {
+	//Get Internal token endpoint for the environment
+	internalTokenEndpoint := GetInternalTokenEndpointOfEnv(env,filePath)
+	slittedString := strings.Split(internalTokenEndpoint,defaultTokenEndPoint)
+	//Get Apim endpoint or publisher endpoint
+	extractedTokenEndpoint := slittedString[0]
+	return extractedTokenEndpoint + defaultRevokeEndpointSuffix
 }
```
Comment (malinthaprasan) [inline import-export-cli/utils/envManagementUtils.go:367]: Shall we add a new line here?

File: import-export-cli/cmd/logout.go [hunk 2]
```diff
@@ -49,10 +49,18 @@ var logoutCmd = &cobra.Command{
...
+	accessToken, err := credentials.GetOAuthAccessToken(cred, environment)
+	error := credentials.RevokeAccessToken(cred, environment, accessToken)
+	if error != nil {
+		return err
+	}
 	store, err := credentials.GetDefaultCredentialStore()
 	if err != nil {
 		return err
 	}
+	fmt.Println("Logged out from", environment, "environment")
```
Comment (1akshitha) [inline import-export-cli/cmd/logout.go:63]: Don't we need to put a space at the end of the strings before environment?

File: import-export-cli/credentials/credentials.go [hunk 3]
```diff
@@ -85,3 +87,37 @@ func GetOAuthAccessToken(credential Credential, env string) (string, error) {
...
+	// set headers to request
+	headers := make(map[string]string)
+	headers[utils.HeaderContentType] = utils.HeaderValueXWWWFormUrlEncoded
+	headers[utils.HeaderAuthorization] = utils.HeaderValueAuthBasicPrefix + " " + b64EncodedClientIDClientSecret
+
+	//Create body for the request
+	body := utils.HeaderToken + token + utils.TokenTypeForRevocation
+
+	resp, err := utils.InvokePOSTRequest(tokenRevokeEndpoint, headers, body)
+	utils.Logln(utils.LogPrefixInfo + "connecting to " + tokenRevokeEndpoint)
```
Comment (1akshitha) [inline import-export-cli/credentials/credentials.go:107]: Don't we need to put this log line before the invokePostRequest method? as per the log, it should print before the request originated from the ctl.

File: import-export-cli/utils/envManagementUtils.go [hunk 4]
```diff
@@ -353,4 +353,15 @@ func GetInternalTokenEndpointOfEnv(env, filePath string ) string {
 		internalTokenEndpoint = GetTokenEndPointFromPublisherEndpoint(publisherEndpointOfEnv)
 	}
 	return internalTokenEndpoint
-}
\ No newline at end of file
+}
+
+//Get token endpoint for Token revocation
```
Comment (1akshitha) [inline import-export-cli/utils/envManagementUtils.go:358]: Shall we add another comment explaining the function arguments?

File: import-export-cli/cmd/logout.go (hunk 2 above)
Comment (Chamindu36) [reply import-export-cli/cmd/logout.go:63]: Fixed with c672258dbfed4a533e8a19dc8f1ef5050d4bd02e

File: import-export-cli/credentials/credentials.go (hunk 3 above)
Comment (Chamindu36) [reply import-export-cli/credentials/credentials.go:107]: Fixed with c672258dbfed4a533e8a19dc8f1ef5050d4bd02e

File: import-export-cli/utils/envManagementUtils.go (hunk 4 above)
Comment (Chamindu36) [reply import-export-cli/utils/envManagementUtils.go:358]: Fixed with c672258dbfed4a533e8a19dc8f1ef5050d4bd02e

**Pull request merged**